### PR TITLE
Actually push inventory from mender-update

### PR DESCRIPTION
**Pull request by vpodzime**
TODO:
- [x] extend the new test to check periodic data submission

### CODE REVIEW

File: common/common.hpp [hunk 1]
```diff
@@ -90,6 +90,16 @@ static inline bool MapContainsStringKey(
 	return map.find(str) != map.end();
 }
 
+template <typename KeyType, typename ValueType>
+static inline vector<KeyType> GetMapKeyVector(const unordered_map<KeyType, ValueType> &map) {
+	vector<KeyType> ret;
+	ret.reserve(map.size());
```
Comment (oleorhagen) [inline common/common.hpp:96]: Suggestion: Reserve the size in the initialiser?

File: mender-update/inventory.cpp [hunk 2]
```diff
@@ -105,11 +110,16 @@ error::Error PushInventoryData(
 	// replace the trailing comma with the closing square bracket
 	payload[payload.size() - 1] = ']';
 
+	size_t payload_hash = std::hash<string> {}(payload);
```
Comment (oleorhagen) [inline mender-update/inventory.cpp:113]: Had to look this one up, so we call a function on the constructed object to hash..

**Review by oleorhagen — APPROVED** [review]
Looks good imo

File: mender-update/inventory.cpp [hunk 3]
```diff
@@ -82,16 +85,18 @@ error::Error PushInventoryData(
 	stringstream top_ss;
 	top_ss << "[";
 	auto inv_data = ex_inv_data.value();
-	for (const auto &kv : inv_data) {
+	auto key_vector = common::GetMapKeyVector(inv_data);
+	std::sort(key_vector.begin(), key_vector.end());
```
Comment (kacf) [inline mender-update/inventory.cpp:89]: Could also just switch over to `std::map`, which is sorted from the get-go, unlike `unordered_map`.

**Review by kacf — APPROVED** [review]

File: common/common.hpp (hunk 1 above)
Comment (vpodzime) [reply common/common.hpp:96]: I actually tried that. I did `vector<KeyType> ret(map.size())`. But that doesn't reserve space, it actually creates a vector of that size with default-initialized items.

File: mender-update/inventory.cpp (hunk 2 above)
Comment (vpodzime) [reply mender-update/inventory.cpp:113]: Yeah, it's a bit confusing.

File: mender-update/inventory.cpp (hunk 3 above)
Comment (vpodzime) [reply mender-update/inventory.cpp:89]: Yes, but to me it makes more sense to keep the data unordered and then only sort it in the place where we need it sorted.

**Pull request merged**